addons/eu/en16931/bill.go

```diff
@@ -104,3 +112,61 @@ func validateBillInvoiceTax(value any) error {
 		),
 	)
 }
+
+func validateBillLine(line *bill.Line) error {
+
+	return validation.ValidateStruct(line,
+		validation.Field(&line.Discounts,
+			validation.Each(
+				validation.By(validateBillLineDiscount),
+			),
+			validation.Skip,
+		),
+		validation.Field(&line.Charges,
+			validation.Each(
+				validation.By(validateBillLineCharge),
+			),
+			validation.Skip,
+		),
+	)
+}
+
+func validateBillLineCharge(value any) error {
+	// BR-44
+	charge, ok := value.(*bill.LineCharge)
+	if !ok || charge == nil {
+		return nil
+	}
+	if charge.Reason == "" && (charge.Ext == nil || charge.Ext[untdid.ExtKeyCharge] == "") {
+		return validation.NewError("reason", "either a reason or a charge type extension is required")
```

Comment:
It could be useful to include the `BR-44` instead of `reason` in the New Error call.